### PR TITLE
Fix display of generator

### PR DESCRIPTION
**Pull request by bruchar1**

### CODE REVIEW

File: mesonbuild/build.py [hunk 1]
```diff
@@ -1704,6 +1704,9 @@ class FileInTargetPrivateDir:
     def __init__(self, fname: str):
         self.fname = fname
 
+    def __repr__(self) -> str:
+        return self.fname
```
Comment (eli-schwartz) [inline mesonbuild/build.py:1708]: Seems wrong to misuse repr for this. Didn't we use File in all cases where FileInTargetPrivateDir now gets used? What was the exact display before?

**Review by eli-schwartz — COMMENTED** [review]
Commit message is sparse...

File: mesonbuild/build.py (hunk 1 above)
Comment (eli-schwartz) [reply mesonbuild/build.py:1708]: sole_output is guaranteed to be a string at this stage, even if that string is the string:

```
sole_output = str('<mesonbuild.build.FileMaybeInTargetPrivateDir object at 0x000001C47609D5D0>')
```

Comment (bruchar1) [reply mesonbuild/build.py:1708]: I see... Therefore this is `__str__` that is needed for `FileInTargetPrivateDir`.

Comment (bruchar1) [reply mesonbuild/build.py:1708]: @eli-schwartz Is it good for you?

**Comment by bruchar1** [comment]
Was probably caused by #11654

**Pull request merged**